examples/reserved-ips/main.tf

```diff
@@ -0,0 +1,97 @@
+  resource_group_name          = var.resource_group == null ? "${var.prefix}-resource-group" : null
+  existing_resource_group_name = var.resource_group
+}
+
+##############################################################################
+# Create a VPC for this example using defaults from terraform-ibm-landing-zone-vpc
+# ( 3 subnets across the 3 AZs in the region )
+##############################################################################
+
+module "vpc" {
+  count             = var.vpc_id != null ? 0 : 1
+  source            = "terraform-ibm-modules/landing-zone-vpc/ibm"
+  version           = "7.5.0"
+  resource_group_id = module.resource_group.resource_group_id
+  region            = var.region
+  prefix            = var.prefix
+  name              = var.vpc_name
+  tags              = var.resource_tags
+}
+
+##############################################################################
+# Demonstrate how to create a custom security group that is applied to the VPEs
+# This examples allow all workload associated with the default VPC security group
+# to interact with the VPEs
+##############################################################################
+
+data "ibm_is_vpc" "vpc" {
+  # Explicit depends as the vpc_name is known prior to VPC creation
+  depends_on = [
+    module.vpc
+  ]
+  name = var.vpc_id != null ? var.vpc_id : module.vpc[0].vpc_name
+}
+
+data "ibm_is_security_group" "default_sg" {
+  name = data.ibm_is_vpc.vpc.default_security_group_name
+}
+
+module "vpe_security_group" {
+  source                       = "terraform-ibm-modules/security-group/ibm"
+  version                      = "2.0.0"
+  security_group_name          = "${var.prefix}-vpe-sg"
+  add_ibm_cloud_internal_rules = false # No need for the internal ibm cloud rules for SG associated with VPEs
+
+  security_group_rules = [{
+    name      = "allow-all-default-sg-inbound"
+    direction = "inbound"
+    remote    = data.ibm_is_security_group.default_sg.id
+  }]
+
+  resource_group = module.resource_group.resource_group_id
+  vpc_id         = var.vpc_id != null ? var.vpc_id : module.vpc[0].vpc_id
+}
+
+##############################################################################
+# Create Reserved IPs in the VPC
+##############################################################################
+module "ips" {
+  source           = "../../modules/reserved-ips"
+  region           = var.region
+  subnet_zone_list = var.vpc_id != null ? var.subnet_zone_list : module.vpc[0].subnet_zone_list
+  reserved_ips     = {}
+  cloud_services = [
+    {
+      service_name = "kms"
+    },
+    {
+      service_name = "cloud-object-storage"
+    }
+  ]
+}
+
+##############################################################################
+# Create VPEs in the VPC
+##############################################################################
+module "vpes" {
+  depends_on         = [module.ips]
```

Comment:
Try to always avoid using explicit `depends_on` - it messes up the dependency tree which can come back to haunt us in the future (see more info [here](https://developer.hashicorp.com/terraform/language/meta-arguments/depends_on#processing-and-planning-consequences)).

In your case I see no need for it, as there is already an implicit dependency below:
`reserved_ips = module.ips.reserved_ip_map`